Yosemite/Yosemite/Stores/Order/OrdersUpsertUseCase.swift

```diff
@@ -184,4 +186,29 @@ struct OrdersUpsertUseCase {
             }
         }
     }
+
+    /// Updates, inserts, or prunes the provided StorageOrderItem's taxes using the provided read-only OrderItem
+    ///
+    private func handleShippingLineTaxes(_ readOnlyItem: Networking.ShippingLine, _ storageItem: Storage.ShippingLine, _ storage: StorageType) {
+        let shippingID = readOnlyItem.shippingID
+
+        // Upsert the taxes from the read-only orderItem
+        for readOnlyTax in readOnlyItem.taxes {
+            if let existingStorageTax = storage.loadShippingLineTax(shippingID: shippingID, taxID: readOnlyTax.taxID) {
+                existingStorageTax.update(with: readOnlyTax)
+            } else {
+                let newStorageTax = storage.insertNewObject(ofType: Storage.ShippingLineTax.self)
+                newStorageTax.update(with: readOnlyTax)
+                storageItem.addToTaxes(newStorageTax)
+            }
+        }
+
+        // Now, remove any objects that exist in storageOrderItem.taxes but not in readOnlyOrderItem.taxes
+        storageItem.taxes?.forEach { storageTax in
+            if readOnlyItem.taxes.first(where: { $0.taxID == storageTax.taxID } ) == nil {
```

Comment:
could more than one `ShippingLineTax` with the same tax ID exist for different shipping lines? if so, we might want to also check on `$0.shipping.shippingID == readOnlyItem.shippingID` to be sure

Reply:
I think it's possible but:
- We don't support multiple shipping lines at the moment 
- I think those shipping line taxes would be different objects so we don't risk deleting them